src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleExtensions.scala

```diff
@@ -197,6 +197,31 @@ class GoogleExtensions(val directoryDAO: DirectoryDAO, val accessPolicyDAO: Acce
     } yield ()
   }
 
+
+  //this class is getting to be too big. look into moving some things out of it.
+  def getPetServiceAccountKey(userId: WorkbenchUserId, project: GoogleProject): Future[ServiceAccountKey] = {
+    for {
+      maybePet <- directoryDAO.loadPetServiceAccount(PetServiceAccountId(userId, project))
+      response <- maybePet match {
+        case Some(pet) =>
+          googleIamDAO.createServiceAccountKey(project, pet.serviceAccount.email) recover {
+            case e: GoogleJsonResponseException if e.getDetails.getCode == StatusCodes.TooManyRequests.intValue => throw new WorkbenchException("You have reached the 10 key limit on service accounts. Please remove one to create another.")
+          }
+        case None => throw new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.NotFound, "Pet service account not found. Please create one first and then try again."))
+      }
+    } yield response
+  }
+
+  def removePetServiceAccountKey(userId: WorkbenchUserId, project: GoogleProject, keyId: ServiceAccountKeyId): Future[Unit] = {
+    for {
+      maybePet <- directoryDAO.loadPetServiceAccount(PetServiceAccountId(userId, project))
+      response <- maybePet match {
+        case Some(pet) => googleIamDAO.removeServiceAccountKey(project, pet.serviceAccount.email, keyId)
+        case None => throw new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.NotFound, "Pet service account not found, therefore there is no key to delete"))
```

Comment:
I wouldn't throw an exception here either, if you want we can return a boolean... what does googleIamDAO.removeServiceAccountKey return if the keyId does not exist?

Reply:
`googleIamDAO.removeServiceAccountKey` doesn't seem to care if the keyId doesn't exist. returns unit